### DISCUSSION TITLE
Group member update access level

### DISCUSSION THREAD

**Discussion by icepick4** [python-gitlab/python-gitlab#2745] [Q&A] [answered]
Hello,

I can't update group member access level.

Here is what I already did : 

I tried with that code : 

```py
member = group.members_all.get(group_member.id)
member.access_level=40
member.save()
```

But I get : 'GroupMemberAll' object has no attribute 'save'

Then I saw this [pull request](https://github.com/python-gitlab/python-gitlab/pull/1827)

My question is, how can I update a group member access level, from this list : `group_members_list=group.members_all.list(get_all=True)`
My initial aim is to update every access level member of a group and his sub groups recursively.

Thank's for your help!

**Comment by JohnVillalovos** [comment, marked as answer]
`members_all` finds all members of all groups under the top-level group. So it doesn't have a way to then modify the member because it can be in possibly any group underneath the top-level group.

If you do:
```
member = group.members.get(group_member.id)
member.access_level=40
member.save()
```

That *might* work, if the member is in `group`.

If you want to update the membership for every member in every group then you will have to iterate through all the sub-groups. Using something like `subgroups = groups.subgroups.list()`


https://docs.gitlab.com/ee/api/members.html
